README_python.md

```diff
@@ -0,0 +1,273 @@
+        not syntactically acceptable in Python.
+
+-   Some classes in the Halide API aren't provided because standard Python
+    idioms are a better fit:
+
+    -   `Halide::Tuple` doesn't exist in the Python bindings; an ordinary Python
+        tuple of `Halide::Expr` is used instead.
+    -   `Halide::Realization` doesn't exist in the Python bindings; an ordinary
+        Python tuple of `Halide::Buffer` is used instead.
+
+-   static and instance method overloads with the same name in the same class
+    aren't allowed, so some convenience methods are missing from `Halide::Var`
+
+-   Templated types (notably `Halide::Buffer<>` and `Halide::Param<>`) aren't
+    provided, for obvious reasons; only the equivalents of
+    `Halide::Buffer<void>` and `Halide::Param<void>` are supported.
+
+-   The functions in `Halide::ConciseCasts` are present in the toplevel Halide
+    module in Python, rather than a submodule: e.g., use `halide.i8_sat()`, not
+    `halide.ConciseCasts.i8_sat()`.
+
+-   Only things in the `Halide` namespace are supported; classes and methods
+    that involve using the `Halide::Internal` namespace are not provided.
+
+-   No mechanism is provided for overriding any runtime functions from Python.
+
+-   No mechanism is provided for supporting `Func::define_extern`.
+
+-   `Buffer::for_each_value()` isn't supported yet.
+
+-   `Func::in` becomes `Func.in_` because `in` is a Python keyword.
+
+-   `Func::async` becomes `Func.async_` because `async` is a Python keyword.
+
+-   The `not` keyword cannot be used to negate boolean Halide expressions.
+    Instead, the `logical_not` function can be used and is equivalent to using
+    `operator!` in C++.
+
+-   There is no way to override the logical `and`/`or` operators in Python to
+    work with `Expr`: you must use the bitwise `|` and `&` instead. (Note that
+    incorrectly using using `and`/`or` just short-circuits weirdly, rather than
+    failing with some helpful error; this is an issue that we have not yet found
+    any way to improve, unfortunately.)
+
+-   Some error messages need to be made more informative.
+
+-   Some exceptions are the "incorrect" type (compared to C++ expectations).
+
+-   Many hooks to override runtime functions (e.g. Func::set_error_handler)
+    aren't yet implemented.
+
+-   The following parts of the Halide public API are currently missing entirely
+    from the Python bindings (but are all likely to be supported at some point
+    in the future):
+
+    -   `DeviceInterface`
+    -   `evaluate()`
+
+-   The following parts of the Halide public API are currently only partially
+    supported by the Python bindings (but are all expected to be supported at
+    some point):
+
+    -   `DeviceInterface`
+    -   `evaluate()`
+
+-   The following parts of the Halide public API are currently only partially
+    supported by the Python bindings (but are all expected to be supported at
+    some point):
+
+    -   `ParamMap`
```

Comment:
Merge these together?

Reply:
Actually, the ParamMap stuff should just go away entirely in Python, since it's there only for thread safety, and Callable provides a better way to do that. Willfix.

Reply:
Done